### ISSUE TITLE
caBundle is required in kubeconfig

### ISSUE THREAD

**Issue by aurelia-sl** [kubedb/project#309]
Currently the deploy script fails mid-deploy when no CA bundle is set in Kubeconfig. This is for instance the case if you setup rancher with ACME/LetsEncrypt support.

```
error: error validating "STDIN": error validating data: [ValidationError(ValidatingWebhookConfiguration.webhooks[0].clientConfig): missing required field "caBundle" in io.k8s.api.admissionregistration.v1beta1.WebhookClientConfig, ValidationError(ValidatingWebhookConfiguration.webhooks[1].clientConfig): missing required field "caBundle" in io.k8s.api.admissionregistration.v1beta1.WebhookClientConfig, ValidationError(ValidatingWebhookConfiguration.webhooks[2].clientConfig): missing required field "caBundle" in io.k8s.api.admissionregistration.v1beta1.WebhookClientConfig, ValidationError(ValidatingWebhookConfiguration.webhooks[3].clientConfig): missing required field "caBundle" in io.k8s.api.admissionregistration.v1beta1.WebhookClientConfig, ValidationError(ValidatingWebhookConfiguration.webhooks[4].clientConfig): missing required field "caBundle" in io.k8s.api.admissionregistration.v1beta1.WebhookClientConfig, ValidationError(ValidatingWebhookConfiguration.webhooks[5].clientConfig): missing required field "caBundle" in io.k8s.api.admissionregistration.v1beta1.WebhookClientConfig, ValidationError(ValidatingWebhookConfiguration.webhooks[6].clientConfig): missing required field "caBundle" in io.k8s.api.admissionregistration.v1beta1.WebhookClientConfig, ValidationError(ValidatingWebhookConfiguration.webhooks[7].clientConfig): missing required field "caBundle" in io.k8s.api.admissionregistration.v1beta1.WebhookClientConfig]; if you choose to ignore these errors, turn validation off with --validate=false
```

**Comment by tamalsaha** [comment]
This is a required field by Kubernetes. https://github.com/kubernetes/api/blob/master/admissionregistration/v1beta1/types.go#L290

If you are using Let's Encrypted issued trusted cert with your kube-apiserver, provide the https://letsencrypt.org/certificates/ as your CA bundle.

**Comment by sascha-sl** [comment]
It is required if you use admission, but not in kubeconfig. 

Assuming it's there is the bug (possibly in your tooling rather than kubedb). It's okay to refuse operation if it's not given, it just shouldn't create half the resources and then crash.

**Comment by gridworkz** [comment]
Got the same issue - @tamalsaha how did you resolve this? Cheers, Dave

**Comment by tamalsaha** [comment]
See my reply here: https://github.com/kubedb/project/issues/309#issuecomment-427627415

You can use the CA cert from LE

**Comment by gridworkz** [comment]
Cool, I've got my LE certs. I'm new to KubeDB - I was wondering what file I put them in (or is there a command-line switch I need to turn on when I install KubeDB Operator).

**Comment by gridworkz** [comment]
I'm almost there - I get a timeout when waiting for KubeDB to be ready:

ubuntu@ip-172-31-26-61:~$ curl -fsSL https://raw.githubusercontent.com/kubedb/cli/0.9.0-beta.1/hack/deploy/kubedb.sh \
>     | bash -s -- --enable-validating-webhook=false --enable-mutating-webhook=false --rbac
checking kubeconfig context
cooby-kube

checking whether extended apiserver feature is enabled

KUBEDB_CATALOG=all
KUBEDB_DOCKER_REGISTRY=kubedb
KUBEDB_ENABLE_ANALYTICS=true
KUBEDB_ENABLE_APISERVER=false
KUBEDB_ENABLE_MUTATING_WEBHOOK=false
KUBEDB_ENABLE_RBAC=true
KUBEDB_ENABLE_STATUS_SUBRESOURCE=true
KUBEDB_ENABLE_VALIDATING_WEBHOOK=false
KUBEDB_IMAGE_PULL_POLICY=IfNotPresent
KUBEDB_IMAGE_PULL_SECRET=
KUBEDB_NAMESPACE=kube-system
KUBEDB_OPERATOR_NAME=operator
KUBEDB_OPERATOR_TAG=0.9.0-beta.1
KUBEDB_PURGE=0
KUBEDB_RUN_ON_MASTER=0
KUBEDB_SERVICE_ACCOUNT=kubedb-operator
KUBEDB_UNINSTALL=0

Wrote ca certificates in  /home/ubuntu
Wrote server certificates in  /home/ubuntu
deployment.apps/kubedb-operator created
secret/kubedb-server-cert created
service/kubedb-operator created
apiservice.apiregistration.k8s.io/v1alpha1.validators.kubedb.com created
apiservice.apiregistration.k8s.io/v1alpha1.mutators.kubedb.com created
serviceaccount/kubedb-operator created
clusterrole.rbac.authorization.k8s.io/kubedb-operator reconciled
clusterrolebinding.rbac.authorization.k8s.io/kubedb-operator reconciled
rolebinding.rbac.authorization.k8s.io/kubedb-server-extension-server-authentication-reader reconciled
clusterrolebinding.rbac.authorization.k8s.io/kubedb-server-auth-delegator reconciled
clusterrole.rbac.authorization.k8s.io/kubedb:core:admin reconciled
clusterrole.rbac.authorization.k8s.io/kubedb:core:edit reconciled
clusterrole.rbac.authorization.k8s.io/kubedb:core:view reconciled

waiting until kubedb operator deployment is ready
timed out waiting for the condition
KubeDB operator deployment failed to be ready

I'm just chasing the error down. Have you seen this before? I'm running a fresh Rancher2.10/Kubernetes install on Ubuntu 16.04.

**Comment by tamalsaha** [comment]
Closing old issue. Please file a new issue if the problem persists.

**Comment by tamalsaha** [comment]
This was fixed in 1.13 release in upstream.
